src/pydicom/pixels/utils.py

```diff
@@ -2098,3 +2168,264 @@ def unpack_bits(src: bytes, as_array: bool = True) -> "np.ndarray | bytes":
         raise ValueError("unpack_bits() requires NumPy if 'as_array = True'")
 
     return b"".join(map(_UNPACK_LUT.__getitem__, src))
+
+
+def get_packed_frame(
+    src: bytes,
+    index: int,
+    frame_length: int,
+    pad: bool = True,
+) -> bytes:
+    """Retrieve a single bit-packed frame from a bit-packed buffer.
+
+    .. versionadded:: 3.1
+
+    This should only be used for natively-encoded single bit data (*Bits
+    Allocated* = 1) in a "bit-packed" representation (one byte containing 8
+    pixels). The pixel data of such images may have pixel data from multiple
+    frames within a single byte.
+
+    See :dcm:`Chapter 5, 8.1.1<part05/chapter_8.html#sect_8.1.1>`, particularly
+    the note about multi-frame images with *Bits Allocated* = 1.
+
+    Parameters
+    ----------
+    src : bytes
+        The natively encoded bit-packed representation of a full *Pixel Data*
+        element value, potentially including multiple frames.
+    index : int
+        Non-negative, zero-based index of the frame to extract.
+    frame_length : int
+        Number of pixels in each frame.
+    pad : bool
```

Comment:
Nitpick: `pad : bool, optional`

Reply:
Addressed in [866c50c](https://github.com/pydicom/pydicom/pull/2216/commits/866c50c2a6c9f526c132a49ccd72012c3ea62b33)